prime-router/docs/design/proposals/0023-condition-to-code-mapping/0023-condition-to-code-mapping.md

```diff
@@ -0,0 +1,383 @@
+                },
+                "subject": {
+                    "reference": "Patient/9473889b-b2b9-45ac-a8d8-191f27132912"
+                },
+                "performer": [
+                    {
+                        "reference": "Organization/1a0139b9-fc23-450b-9b6c-cd081e5cea9d"
+                    }
+                ],
+                "valueCodeableConcept": {
+                    "coding": [
+                        {
+                            "system": "http://snomed.info/sct",
+                            "code": "260373001",
+                            "display": "Detected"
+                        }
+                    ]
+                },
+                "meta": {
+                    "tag": [
+                          {
+                              "code": "541000000000000",
+                              "display": "Infection caused by novel Influenza A virus variant (disorder)"
+                         }
+                    ]
+                }
+            }
+        }
+```
+
+In order to account for both HL7 and FHIR input from senders, the lookup should occur after conversion of the incoming HL7 V2 message into a FHIR bundle. Since the sender transforms are used after conversion from HL7 to FHIR, adding the below elements to the default sender transform will add the condition tag to the observation resource.
+
+```yaml
+  - name: test-condition
+    resource: 'Bundle.entry.resource.ofType(Observation)'
+    condition: '%resource.code.coding.exists()'
+    bundleProperty: '%resource.meta.tag.code'
+    value: ['%resource.code.coding.code']
+    valueSet:
+      lookupTable:
+        tableName: Condition-Mapping
+        keyColumn: code
+        valueColumn: condition_code
+```
+This element will check against the RCTC values loaded into the Observation-Mapping tables.
+
+Not all reportable conditions can be identified from order and result LOINC codes. In the case of bacterial cultures the condition is identified by the found organism which is populated in the result field/resource (OBX-5 or Bundle.entry.resource.ofType(Observation).value.valueCodeableConcept). In order to correctly tag these messages, an additional element will need to be added to the transform like below to check the result code.
+
+```yaml
+- name: result-condition
+  resource: 'Bundle.entry.resource.ofType(Observation)'
+  condition: '%resource.code.coding.exists() and %resource.meta.tag.code.not.exists()'
+  bundleProperty: '%resource.meta.tag.code'
+  value: ['%resource.valueCodeableConcept.coding.code']
+  valueSet:
+    lookupTable:
+      tableName: Condition-Mapping
+      keyColumn: code
+      valueColumn: condition_code
+```
+This element should be placed after the above elements as we only want to check the mapping for a result value if we have failed to find a match in the test ordered and test performed locations. 
```

Comment:
We would have to change the monitoring strategy if we did that. Otherwise we would be getting a lot of false positive for "unmapped" snomed codes when the condition was indicated by the LOINC code. To clarify for a particular set of observations the condition will be indicated by either the LOINC or SNOMED code, not both.